cli/commands/catalog/module/repo.go

```diff
@@ -163,69 +169,164 @@ func (repo *Repo) ModuleURL(moduleDir string) (string, error) {
 	return "", errors.Errorf("hosting: %q is not supported yet", remote.Host)
 }
 
-// clone clones the repository to a temporary directory if the repoPath is URL
-func (repo *Repo) clone(ctx context.Context) error {
+type CloneOptions struct {
+	SourceURL  string
+	TargetPath string
+	Context    context.Context
+	Logger     log.Logger
+}
+
+func (repo *Repo) clone(ctx context.Context, l log.Logger) error {
```

Comment:
For what do you pass `l log.Logger` as arg of the func when you can use `repo.logger` in all nested funcs?